sycl/plugins/level_zero/usm_allocator.cpp

```diff
@@ -673,6 +686,14 @@ void *Slab::getEnd() const {
 
 bool Slab::hasAvail() { return NumAllocated != getNumChunks(); }
 
+// If a slab was available in the pool then note that the current pooled
+// size has reduced by the size of this slab.
+void Bucket::decrementPool(bool &FromPool) {
+  FromPool = true;
```

Comment:
What does `FromPool` mean and what is the reason set it here (as opposed to setting it to true in the caller)?

Reply:
The flag FromPool tells the caller (at the outermost level) that this allocation came out of the pool and is not a fresh USM allocation. The function decrementPool updates pool usage statistics and also sets the output flag FromPool to indicate an allocation is being pulled out of the pool. The reason to set FromPool to true within the function is to avoid duplicating the same assignment in all the places decrementPool is called from.